src/Features/CSharp/Portable/Diagnostics/Analyzers/CSharpUseImplicitTypeDiagnosticAnalyzer.cs

```diff
@@ -262,6 +258,19 @@ protected override bool TryAnalyzeVariableDeclaration(TypeSyntax typeName, Seman
             return declaredType.Equals(initializerType);
         }
 
+        private static bool IsExpressionSameAfterVarConversion(
+            ExpressionSyntax expression,
+            SemanticModel semanticModel,
+            CancellationToken cancellationToken)
+        {
+            // Get the conversion that occurred between the expression's type and type implied by the expression's context
+            // and filter out implicit conversions. If an implicit conversion (other than identity) exists
+            // and if we're replacing the declaration with 'var' we'd be changing the semantics by inferring type of
+            // initializer expression and thereby losing the conversion.
+            var conversion = semanticModel.GetConversion(expression, cancellationToken);
+            return conversion.Exists && conversion.IsImplicit && conversion.IsIdentity;
```

Comment:
:bulb: `conversion.IsIdentity` implies the other two, so this can be simplified to only check the one value.